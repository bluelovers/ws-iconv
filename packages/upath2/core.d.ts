import types, { IPathNode, IPath, IParse, ORIGIN_KEY, IPathType } from './lib/type';
export { types, IPathNode, IPath, IParse, IPathType };
export declare class PathWrap implements IPath {
    sep: string;
    name: string;
    delimiter: string;
    win32: IPath;
    posix: IPath;
    upath: PathWrap;
    default: PathWrap;
    fn: IPath;
    [ORIGIN_KEY]?: IPathType;
    constructor(path: any, id: string);
    join<T = string, U = string>(path: T, ...paths: U[]): string;
    normalize<T = string>(path: T): string;
    relative<T = string, U = string>(from: T, to: U): string;
    resolve<T = string, U = string>(path: T, ...paths: U[]): string;
    parse<T = string>(path: T): IParse;
    format<T = IParse>(pathObject: T): string;
    basename<T = string, U = string>(path: T, ext?: U): string;
    dirname<T = string>(path: T): string;
    extname<T = string>(path: T): string;
    isAbsolute<T = string>(path: T): boolean;
}
export declare namespace PathWrap {
    let fn: types.IPath;
    type IPath = types.IPath;
    type IPathNode = types.IPathNode;
    type IParse = types.IParse;
}
export declare const posix: types.IPath;
export declare const win32: types.IPath;
export declare type IUPath = PathWrap & {
    default: IUPath;
    upath: IUPath;
    PathWrap: typeof PathWrap;
};
export declare const upath: IUPath;
export declare const fn: types.IPath;
export default upath;
