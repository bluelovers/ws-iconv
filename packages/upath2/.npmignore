# Created by .ignore support plugin (hsz.mobi)
### Node template
# Logs
logs
*.log
npm-debug.log*

# Runtime data
pids
*.pid
*.seed

# Directory for instrumented libs generated by jscoverage/JSCover
lib-cov

# Coverage directory used by tools like istanbul
coverage

# Grunt intermediate storage (http://gruntjs.com/creating-plugins#storing-task-files)
.grunt

# node-waf configuration
.lock-wscript

# Compiled binary addons (http://nodejs.org/api/addons.html)
build/Release

# Dependency directory
# https://docs.npmjs.com/misc/faq#should-i-check-my-node-modules-folder-into-git
node_modules
### Example user template template
### Example user template

# IntelliJ project files
.idea
*.iml
out
gen
/test/
/node_modules/
/.idea/
### jsonly template
*.ts
!*.d.ts
### jsonly_bin template
bin/*.ts
bin/*.d.ts
/yarn.lock
/tsconfig.json
### npm2 template

~ci.list.txt
~ci.log.txt
~ci.errors.txt

*.stackdump
*.bak
*.old

tsconfig.json
package-lock.json

test

.github
.gitkeep

/.*

tests



/~*

__test__
__tests__

**/node_modules/
