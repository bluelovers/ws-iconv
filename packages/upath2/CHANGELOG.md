# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [3.0.2](https://github.com/bluelovers/ws-iconv/compare/upath2@3.0.1...upath2@3.0.2) (2020-06-09)

**Note:** Version bump only for package upath2





## 3.0.1 (2020-06-09)

**Note:** Version bump only for package upath2
