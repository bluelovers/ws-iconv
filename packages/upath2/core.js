"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const _path = require("path");
const type_1 = require("./lib/type");
exports.types = type_1.default;
class PathWrap {
    constructor(path, id) {
        this.sep = '/';
        let _static = getStatic(this);
        this[type_1.ORIGIN_KEY] = path;
        this.name = id;
        delete this[id];
        this[id] = this;
        Object.defineProperty(this, type_1.ORIGIN_KEY, { enumerable: false, });
        this.fn = Object.assign(this.__proto__, path, _static.fn);
    }
    join(path, ...paths) {
        return _this_origin(this).join(path, ...paths).replace(/\\/g, this.sep);
    }
    normalize(path) {
        return _this_origin(this).normalize(path).replace(/\\/g, this.sep);
    }
    relative(from, to) {
        return _this_origin(this).relative(from.toString(), to.toString()).replace(/\\/g, this.sep);
    }
    resolve(path, ...paths) {
        return _this_origin(this).resolve(path, ...paths).replace(/\\/g, this.sep);
    }
    parse(path) {
        let ret = _this_origin(this).parse(path);
        ret.root = ret.root.replace(/\\/g, this.sep);
        ret.dir = ret.dir.replace(/\\/g, this.sep);
        return ret;
    }
    format(pathObject) {
        return _replace_sep(this, _this_origin(this).format(pathObject));
    }
    basename(path, ext) {
        return _this_origin(this).basename(path, ext);
    }
    dirname(path) {
        return _this_origin(this).dirname(path);
    }
    extname(path) {
        return _this_origin(this).extname(path);
    }
    isAbsolute(path) {
        return _this_origin(this).isAbsolute(path);
    }
}
exports.PathWrap = PathWrap;
(function (PathWrap) {
    let __proto__ = {};
    for (let i in Object.getOwnPropertyDescriptors(PathWrap.prototype)) {
        __proto__[i] = PathWrap.prototype[i];
    }
    PathWrap.fn = __proto__;
    PathWrap.fn['fn'] = PathWrap.fn;
    PathWrap.fn.sep = '/';
    PathWrap.prototype.fn = PathWrap.fn;
})(PathWrap = exports.PathWrap || (exports.PathWrap = {}));
function getStatic(who) {
    return who.__proto__.constructor;
}
exports.posix = new PathWrap(_path.posix, 'posix');
exports.win32 = new PathWrap(_path.win32, 'win32');
const _upath = new PathWrap(_path, 'upath');
exports.upath = _upath;
exports.upath.PathWrap = PathWrap;
exports.upath.fn.win32 = exports.win32;
exports.upath.fn.posix = exports.posix;
exports.upath.fn.upath = exports.upath;
exports.upath.fn.default = exports.upath;
exports.fn = PathWrap.fn = exports.upath.fn;
_path.upath = exports.upath;
for (let key of [
    'win32',
    'posix',
    'upath',
]) {
    exports.win32.fn[key] = exports.posix.fn[key] = exports.upath.fn[key] = exports.upath[key];
}
exports.win32.default = exports.posix.default = exports.upath.default = exports.upath;
exports.default = exports.upath;
function _this_origin(who) {
    if (who[type_1.ORIGIN_KEY]) {
        return who[type_1.ORIGIN_KEY];
    }
    else if (who === exports.upath) {
        return _path;
    }
    else if (who === exports.win32) {
        return _path.win32;
    }
    else if (who === exports.posix) {
        return _path.posix;
    }
    throw new TypeError(`this not PathWrap`);
}
function _replace_sep(who, input) {
    return input.replace(/\\/g, who.sep);
}
Object.freeze(exports);
