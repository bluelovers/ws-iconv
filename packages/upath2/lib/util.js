"use strict";
/**
 * Created by user on 2020/6/9.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.getStatic = exports._replace_sep = exports._strip_sep = void 0;
const tslib_1 = require("tslib");
const path_is_network_drive_1 = (0, tslib_1.__importDefault)(require("path-is-network-drive"));
const path_strip_sep_1 = (0, tslib_1.__importDefault)(require("path-strip-sep"));
exports._strip_sep = path_strip_sep_1.default;
function _replace_sep(who, input) {
    let sep = who.sep;
    if (who.name !== 'posix' && (0, path_is_network_drive_1.default)(input)) {
        sep = '\\';
        input = sep + sep + input
            .slice(2)
            .replace(/[/\\]/g, sep);
    }
    else {
        input = input
            .replace(/[/\\]/g, sep);
    }
    //	if (/^\w:[/\\]$/.test(input))
    //	{
    //		return input
    //	}
    return input;
}
exports._replace_sep = _replace_sep;
function getStatic(who) {
    return who.__proto__.constructor;
}
exports.getStatic = getStatic;
//# sourceMappingURL=util.js.map