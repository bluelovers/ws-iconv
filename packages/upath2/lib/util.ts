/**
 * Created by user on 2020/6/9.
 */

import { IPath } from './type';
import { PathWrap } from '../core';
import pathIsNetworkDrive from 'path-is-network-drive';
import _strip_sep from 'path-strip-sep';

export { _strip_sep }

export function _replace_sep(who: Pick<IPath, 'sep' | 'name'>, input: string): string
{
	let sep = who.sep;

	if (who.name !== 'posix' && pathIsNetworkDrive(input))
	{
		sep = '\\';

		input = sep + sep + input
			.slice(2)
			.replace(/[/\\]/g, sep)
		;
	}
	else
	{
		input = input
			.replace(/[/\\]/g, sep)
		;
	}

//	if (/^\w:[/\\]$/.test(input))
//	{
//		return input
//	}

	return input
		//.replace(/([^/\\])[/\\]$/, '$1')
		;
}

export function getStatic(who): PathWrap
{
	return who.__proto__.constructor;
}

export function defaults(destination, ...input)
{
	destination = destination || {};

	input.forEach(defaults => {
		for (const key in defaults) {
			if (defaults.hasOwnProperty(key) && !destination.hasOwnProperty(key)) {
				destination[key] = defaults[key];
			}
		}
	});
	  
	return destination;
}
