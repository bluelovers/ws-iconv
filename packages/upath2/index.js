"use strict";
const core_1 = require("./core");
require("./lib/fs");
const path = core_1.upath;
module.exports = path;
