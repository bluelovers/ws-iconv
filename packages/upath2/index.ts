/**
 * Created by user on 2017/12/10/010.
 */


import { upath, PathWrap, fn, IUPath } from './core';
import * as _upath from './core';

import './lib/fs';
import types, { IPath, IPathNode, IParse } from './lib/type';

/*
const path = upath as typeof upath & {
	default: typeof upath,
};

path.default = path;
*/
const path = upath;
export = path;
