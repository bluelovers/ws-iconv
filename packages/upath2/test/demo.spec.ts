import upath2 from '..';
import * as upath from 'upath2';
import testCases, { ITestCaseRow } from './data/testcases';
import { inspect } from 'util';

describe(`check`, () =>
{

	_setup(`upath2`, upath2);

	([
		'upath',
		'win32',
		//'posix',
	] as Extract<keyof typeof upath2, string>[])
		.forEach(name => {

			const path = upath2[name] as any as typeof upath2;

			_setup(name, path);

		})
	;

})

function _setup(name: string, path: typeof upath2)
{
	describe(name, () =>
	{

		Object.entries(testCases)
			.forEach(([name, testCases]) => {

				let fn = path[name].bind(path);

				_test(fn, name, testCases);

				fn = path[name];

				_test(fn, `(unbind) ${name}`, testCases);

			})
		;

	})

	function _test(fn, name: string, testCases: ITestCaseRow[])
	{
		describe(name, () =>
		{
			testCases
				.forEach(testCase =>
				{
					testCase && test(inspect(testCase[0]), () => {
						let actual;
						let expected
						let expected2

						if (typeof testCase[0] === 'string')
						{
							actual = fn(testCase[0])
							expected2 = upath[name]?.(testCase[0])
							expected = testCase[1]

						}
						else
						{
							actual = fn(...testCase[0])
							expected2 = upath[name]?.(...testCase[0])
							expected = testCase[1]
						}

						if (typeof expected !== "undefined")
						{
							expect(actual).toStrictEqual(expected)
						}

						console.dir({
							upath2: actual,
							upath: expected2,
						})

						expect(actual).toMatchSnapshot();
					});
				})
			;

		});
	}
}

test(`\\`, () => {

	let c = `\\`

	expect(upath2.normalize(c)).toHaveLength(1)

	expect(upath2.dirname(c)).toHaveLength(1)

})

test(`/`, () => {

	let c = `/`

	expect(upath2.normalize(c)).toHaveLength(1)

	expect(upath2.dirname(c)).toHaveLength(1)

})
