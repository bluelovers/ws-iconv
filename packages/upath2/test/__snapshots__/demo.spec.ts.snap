// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check upath (unbind) join [ 'some/nodejs/deep', '../path' ] 1`] = `"some/nodejs/path"`;

exports[`check upath (unbind) join [ 'some/nodejs\\\\windows', '../path' ] 1`] = `"some/nodejs/path"`;

exports[`check upath (unbind) join [ 'some\\\\windows\\\\only', '..\\\\path' ] 1`] = `"some/windows/path"`;

exports[`check upath (unbind) normalize '//\\\\windows\\\\..\\\\unix/mixed/' 1`] = `"/unix/mixed/"`;

exports[`check upath (unbind) normalize '/root/of/unix/somefile.ext' 1`] = `"/root/of/unix/somefile.ext"`;

exports[`check upath (unbind) normalize 'c:/windows/../nodejs/path' 1`] = `"c:/nodejs/path"`;

exports[`check upath (unbind) normalize 'c:/windows/nodejs/path' 1`] = `"c:/windows/nodejs/path"`;

exports[`check upath (unbind) normalize 'c:\\\\Windows\\\\Directory\\\\somefile.ext' 1`] = `"c:/Windows/Directory/somefile.ext"`;

exports[`check upath (unbind) normalize 'c:\\\\windows\\\\..\\\\nodejs\\\\path' 1`] = `"c:/nodejs/path"`;

exports[`check upath (unbind) normalize 'c:\\\\windows\\\\nodejs\\\\path' 1`] = `"c:/windows/nodejs/path"`;

exports[`check upath (unbind) parse '//\\\\windows\\\\..\\\\unix/mixed/' 1`] = `
Object {
  "base": "mixed",
  "dir": "/unix",
  "ext": "",
  "name": "mixed",
  "root": "/",
}
`;

exports[`check upath (unbind) parse '//windows\\\\unix/mixed' 1`] = `
Object {
  "base": "mixed",
  "dir": "\\\\\\\\windows\\\\unix\\\\",
  "ext": "",
  "name": "mixed",
  "root": "\\\\\\\\windows\\\\unix\\\\",
}
`;

exports[`check upath (unbind) parse '/root/of/unix/somefile.ext' 1`] = `
Object {
  "base": "somefile.ext",
  "dir": "/root/of/unix",
  "ext": ".ext",
  "name": "somefile",
  "root": "/",
}
`;

exports[`check upath (unbind) parse '\\\\windows//unix/mixed' 1`] = `
Object {
  "base": "mixed",
  "dir": "/windows/unix",
  "ext": "",
  "name": "mixed",
  "root": "/",
}
`;

exports[`check upath (unbind) parse 'c:/windows/../nodejs/path' 1`] = `
Object {
  "base": "path",
  "dir": "c:/nodejs",
  "ext": "",
  "name": "path",
  "root": "c:/",
}
`;

exports[`check upath (unbind) parse 'c:/windows/nodejs/path' 1`] = `
Object {
  "base": "path",
  "dir": "c:/windows/nodejs",
  "ext": "",
  "name": "path",
  "root": "c:/",
}
`;

exports[`check upath (unbind) parse 'c:\\\\Windows\\\\Directory\\\\somefile.ext' 1`] = `
Object {
  "base": "somefile.ext",
  "dir": "c:/Windows/Directory",
  "ext": ".ext",
  "name": "somefile",
  "root": "c:/",
}
`;

exports[`check upath (unbind) parse 'c:\\\\windows\\\\..\\\\nodejs\\\\path' 1`] = `
Object {
  "base": "path",
  "dir": "c:/nodejs",
  "ext": "",
  "name": "path",
  "root": "c:/",
}
`;

exports[`check upath (unbind) parse 'c:\\\\windows\\\\nodejs\\\\path' 1`] = `
Object {
  "base": "path",
  "dir": "c:/windows/nodejs",
  "ext": "",
  "name": "path",
  "root": "c:/",
}
`;

exports[`check upath join [ 'some/nodejs/deep', '../path' ] 1`] = `"some/nodejs/path"`;

exports[`check upath join [ 'some/nodejs\\\\windows', '../path' ] 1`] = `"some/nodejs/path"`;

exports[`check upath join [ 'some\\\\windows\\\\only', '..\\\\path' ] 1`] = `"some/windows/path"`;

exports[`check upath normalize '//\\\\windows\\\\..\\\\unix/mixed/' 1`] = `"/unix/mixed/"`;

exports[`check upath normalize '/root/of/unix/somefile.ext' 1`] = `"/root/of/unix/somefile.ext"`;

exports[`check upath normalize 'c:/windows/../nodejs/path' 1`] = `"c:/nodejs/path"`;

exports[`check upath normalize 'c:/windows/nodejs/path' 1`] = `"c:/windows/nodejs/path"`;

exports[`check upath normalize 'c:\\\\Windows\\\\Directory\\\\somefile.ext' 1`] = `"c:/Windows/Directory/somefile.ext"`;

exports[`check upath normalize 'c:\\\\windows\\\\..\\\\nodejs\\\\path' 1`] = `"c:/nodejs/path"`;

exports[`check upath normalize 'c:\\\\windows\\\\nodejs\\\\path' 1`] = `"c:/windows/nodejs/path"`;

exports[`check upath parse '//\\\\windows\\\\..\\\\unix/mixed/' 1`] = `
Object {
  "base": "mixed",
  "dir": "/unix",
  "ext": "",
  "name": "mixed",
  "root": "/",
}
`;

exports[`check upath parse '//windows\\\\unix/mixed' 1`] = `
Object {
  "base": "mixed",
  "dir": "\\\\\\\\windows\\\\unix\\\\",
  "ext": "",
  "name": "mixed",
  "root": "\\\\\\\\windows\\\\unix\\\\",
}
`;

exports[`check upath parse '/root/of/unix/somefile.ext' 1`] = `
Object {
  "base": "somefile.ext",
  "dir": "/root/of/unix",
  "ext": ".ext",
  "name": "somefile",
  "root": "/",
}
`;

exports[`check upath parse '\\\\windows//unix/mixed' 1`] = `
Object {
  "base": "mixed",
  "dir": "/windows/unix",
  "ext": "",
  "name": "mixed",
  "root": "/",
}
`;

exports[`check upath parse 'c:/windows/../nodejs/path' 1`] = `
Object {
  "base": "path",
  "dir": "c:/nodejs",
  "ext": "",
  "name": "path",
  "root": "c:/",
}
`;

exports[`check upath parse 'c:/windows/nodejs/path' 1`] = `
Object {
  "base": "path",
  "dir": "c:/windows/nodejs",
  "ext": "",
  "name": "path",
  "root": "c:/",
}
`;

exports[`check upath parse 'c:\\\\Windows\\\\Directory\\\\somefile.ext' 1`] = `
Object {
  "base": "somefile.ext",
  "dir": "c:/Windows/Directory",
  "ext": ".ext",
  "name": "somefile",
  "root": "c:/",
}
`;

exports[`check upath parse 'c:\\\\windows\\\\..\\\\nodejs\\\\path' 1`] = `
Object {
  "base": "path",
  "dir": "c:/nodejs",
  "ext": "",
  "name": "path",
  "root": "c:/",
}
`;

exports[`check upath parse 'c:\\\\windows\\\\nodejs\\\\path' 1`] = `
Object {
  "base": "path",
  "dir": "c:/windows/nodejs",
  "ext": "",
  "name": "path",
  "root": "c:/",
}
`;

exports[`check upath2 (unbind) join [ 'some/nodejs/deep', '../path' ] 1`] = `"some/nodejs/path"`;

exports[`check upath2 (unbind) join [ 'some/nodejs\\\\windows', '../path' ] 1`] = `"some/nodejs/path"`;

exports[`check upath2 (unbind) join [ 'some\\\\windows\\\\only', '..\\\\path' ] 1`] = `"some/windows/path"`;

exports[`check upath2 (unbind) normalize '//\\\\windows\\\\..\\\\unix/mixed/' 1`] = `"/unix/mixed/"`;

exports[`check upath2 (unbind) normalize '/root/of/unix/somefile.ext' 1`] = `"/root/of/unix/somefile.ext"`;

exports[`check upath2 (unbind) normalize 'c:/windows/../nodejs/path' 1`] = `"c:/nodejs/path"`;

exports[`check upath2 (unbind) normalize 'c:/windows/nodejs/path' 1`] = `"c:/windows/nodejs/path"`;

exports[`check upath2 (unbind) normalize 'c:\\\\Windows\\\\Directory\\\\somefile.ext' 1`] = `"c:/Windows/Directory/somefile.ext"`;

exports[`check upath2 (unbind) normalize 'c:\\\\windows\\\\..\\\\nodejs\\\\path' 1`] = `"c:/nodejs/path"`;

exports[`check upath2 (unbind) normalize 'c:\\\\windows\\\\nodejs\\\\path' 1`] = `"c:/windows/nodejs/path"`;

exports[`check upath2 (unbind) parse '//\\\\windows\\\\..\\\\unix/mixed/' 1`] = `
Object {
  "base": "mixed",
  "dir": "/unix",
  "ext": "",
  "name": "mixed",
  "root": "/",
}
`;

exports[`check upath2 (unbind) parse '//windows\\\\unix/mixed' 1`] = `
Object {
  "base": "mixed",
  "dir": "\\\\\\\\windows\\\\unix\\\\",
  "ext": "",
  "name": "mixed",
  "root": "\\\\\\\\windows\\\\unix\\\\",
}
`;

exports[`check upath2 (unbind) parse '/root/of/unix/somefile.ext' 1`] = `
Object {
  "base": "somefile.ext",
  "dir": "/root/of/unix",
  "ext": ".ext",
  "name": "somefile",
  "root": "/",
}
`;

exports[`check upath2 (unbind) parse '\\\\windows//unix/mixed' 1`] = `
Object {
  "base": "mixed",
  "dir": "/windows/unix",
  "ext": "",
  "name": "mixed",
  "root": "/",
}
`;

exports[`check upath2 (unbind) parse 'c:/windows/../nodejs/path' 1`] = `
Object {
  "base": "path",
  "dir": "c:/nodejs",
  "ext": "",
  "name": "path",
  "root": "c:/",
}
`;

exports[`check upath2 (unbind) parse 'c:/windows/nodejs/path' 1`] = `
Object {
  "base": "path",
  "dir": "c:/windows/nodejs",
  "ext": "",
  "name": "path",
  "root": "c:/",
}
`;

exports[`check upath2 (unbind) parse 'c:\\\\Windows\\\\Directory\\\\somefile.ext' 1`] = `
Object {
  "base": "somefile.ext",
  "dir": "c:/Windows/Directory",
  "ext": ".ext",
  "name": "somefile",
  "root": "c:/",
}
`;

exports[`check upath2 (unbind) parse 'c:\\\\windows\\\\..\\\\nodejs\\\\path' 1`] = `
Object {
  "base": "path",
  "dir": "c:/nodejs",
  "ext": "",
  "name": "path",
  "root": "c:/",
}
`;

exports[`check upath2 (unbind) parse 'c:\\\\windows\\\\nodejs\\\\path' 1`] = `
Object {
  "base": "path",
  "dir": "c:/windows/nodejs",
  "ext": "",
  "name": "path",
  "root": "c:/",
}
`;

exports[`check upath2 join [ 'some/nodejs/deep', '../path' ] 1`] = `"some/nodejs/path"`;

exports[`check upath2 join [ 'some/nodejs\\\\windows', '../path' ] 1`] = `"some/nodejs/path"`;

exports[`check upath2 join [ 'some\\\\windows\\\\only', '..\\\\path' ] 1`] = `"some/windows/path"`;

exports[`check upath2 normalize '//\\\\windows\\\\..\\\\unix/mixed/' 1`] = `"/unix/mixed/"`;

exports[`check upath2 normalize '/root/of/unix/somefile.ext' 1`] = `"/root/of/unix/somefile.ext"`;

exports[`check upath2 normalize 'c:/windows/../nodejs/path' 1`] = `"c:/nodejs/path"`;

exports[`check upath2 normalize 'c:/windows/nodejs/path' 1`] = `"c:/windows/nodejs/path"`;

exports[`check upath2 normalize 'c:\\\\Windows\\\\Directory\\\\somefile.ext' 1`] = `"c:/Windows/Directory/somefile.ext"`;

exports[`check upath2 normalize 'c:\\\\windows\\\\..\\\\nodejs\\\\path' 1`] = `"c:/nodejs/path"`;

exports[`check upath2 normalize 'c:\\\\windows\\\\nodejs\\\\path' 1`] = `"c:/windows/nodejs/path"`;

exports[`check upath2 parse '//\\\\windows\\\\..\\\\unix/mixed/' 1`] = `
Object {
  "base": "mixed",
  "dir": "/unix",
  "ext": "",
  "name": "mixed",
  "root": "/",
}
`;

exports[`check upath2 parse '//windows\\\\unix/mixed' 1`] = `
Object {
  "base": "mixed",
  "dir": "\\\\\\\\windows\\\\unix\\\\",
  "ext": "",
  "name": "mixed",
  "root": "\\\\\\\\windows\\\\unix\\\\",
}
`;

exports[`check upath2 parse '/root/of/unix/somefile.ext' 1`] = `
Object {
  "base": "somefile.ext",
  "dir": "/root/of/unix",
  "ext": ".ext",
  "name": "somefile",
  "root": "/",
}
`;

exports[`check upath2 parse '\\\\windows//unix/mixed' 1`] = `
Object {
  "base": "mixed",
  "dir": "/windows/unix",
  "ext": "",
  "name": "mixed",
  "root": "/",
}
`;

exports[`check upath2 parse 'c:/windows/../nodejs/path' 1`] = `
Object {
  "base": "path",
  "dir": "c:/nodejs",
  "ext": "",
  "name": "path",
  "root": "c:/",
}
`;

exports[`check upath2 parse 'c:/windows/nodejs/path' 1`] = `
Object {
  "base": "path",
  "dir": "c:/windows/nodejs",
  "ext": "",
  "name": "path",
  "root": "c:/",
}
`;

exports[`check upath2 parse 'c:\\\\Windows\\\\Directory\\\\somefile.ext' 1`] = `
Object {
  "base": "somefile.ext",
  "dir": "c:/Windows/Directory",
  "ext": ".ext",
  "name": "somefile",
  "root": "c:/",
}
`;

exports[`check upath2 parse 'c:\\\\windows\\\\..\\\\nodejs\\\\path' 1`] = `
Object {
  "base": "path",
  "dir": "c:/nodejs",
  "ext": "",
  "name": "path",
  "root": "c:/",
}
`;

exports[`check upath2 parse 'c:\\\\windows\\\\nodejs\\\\path' 1`] = `
Object {
  "base": "path",
  "dir": "c:/windows/nodejs",
  "ext": "",
  "name": "path",
  "root": "c:/",
}
`;

exports[`check win32 (unbind) join [ 'some/nodejs/deep', '../path' ] 1`] = `"some/nodejs/path"`;

exports[`check win32 (unbind) join [ 'some/nodejs\\\\windows', '../path' ] 1`] = `"some/nodejs/path"`;

exports[`check win32 (unbind) join [ 'some\\\\windows\\\\only', '..\\\\path' ] 1`] = `"some/windows/path"`;

exports[`check win32 (unbind) normalize '//\\\\windows\\\\..\\\\unix/mixed/' 1`] = `"/unix/mixed/"`;

exports[`check win32 (unbind) normalize '/root/of/unix/somefile.ext' 1`] = `"/root/of/unix/somefile.ext"`;

exports[`check win32 (unbind) normalize 'c:/windows/../nodejs/path' 1`] = `"c:/nodejs/path"`;

exports[`check win32 (unbind) normalize 'c:/windows/nodejs/path' 1`] = `"c:/windows/nodejs/path"`;

exports[`check win32 (unbind) normalize 'c:\\\\Windows\\\\Directory\\\\somefile.ext' 1`] = `"c:/Windows/Directory/somefile.ext"`;

exports[`check win32 (unbind) normalize 'c:\\\\windows\\\\..\\\\nodejs\\\\path' 1`] = `"c:/nodejs/path"`;

exports[`check win32 (unbind) normalize 'c:\\\\windows\\\\nodejs\\\\path' 1`] = `"c:/windows/nodejs/path"`;

exports[`check win32 (unbind) parse '//\\\\windows\\\\..\\\\unix/mixed/' 1`] = `
Object {
  "base": "mixed",
  "dir": "/unix",
  "ext": "",
  "name": "mixed",
  "root": "/",
}
`;

exports[`check win32 (unbind) parse '//windows\\\\unix/mixed' 1`] = `
Object {
  "base": "mixed",
  "dir": "\\\\\\\\windows\\\\unix\\\\",
  "ext": "",
  "name": "mixed",
  "root": "\\\\\\\\windows\\\\unix\\\\",
}
`;

exports[`check win32 (unbind) parse '/root/of/unix/somefile.ext' 1`] = `
Object {
  "base": "somefile.ext",
  "dir": "/root/of/unix",
  "ext": ".ext",
  "name": "somefile",
  "root": "/",
}
`;

exports[`check win32 (unbind) parse '\\\\windows//unix/mixed' 1`] = `
Object {
  "base": "mixed",
  "dir": "/windows/unix",
  "ext": "",
  "name": "mixed",
  "root": "/",
}
`;

exports[`check win32 (unbind) parse 'c:/windows/../nodejs/path' 1`] = `
Object {
  "base": "path",
  "dir": "c:/nodejs",
  "ext": "",
  "name": "path",
  "root": "c:/",
}
`;

exports[`check win32 (unbind) parse 'c:/windows/nodejs/path' 1`] = `
Object {
  "base": "path",
  "dir": "c:/windows/nodejs",
  "ext": "",
  "name": "path",
  "root": "c:/",
}
`;

exports[`check win32 (unbind) parse 'c:\\\\Windows\\\\Directory\\\\somefile.ext' 1`] = `
Object {
  "base": "somefile.ext",
  "dir": "c:/Windows/Directory",
  "ext": ".ext",
  "name": "somefile",
  "root": "c:/",
}
`;

exports[`check win32 (unbind) parse 'c:\\\\windows\\\\..\\\\nodejs\\\\path' 1`] = `
Object {
  "base": "path",
  "dir": "c:/nodejs",
  "ext": "",
  "name": "path",
  "root": "c:/",
}
`;

exports[`check win32 (unbind) parse 'c:\\\\windows\\\\nodejs\\\\path' 1`] = `
Object {
  "base": "path",
  "dir": "c:/windows/nodejs",
  "ext": "",
  "name": "path",
  "root": "c:/",
}
`;

exports[`check win32 join [ 'some/nodejs/deep', '../path' ] 1`] = `"some/nodejs/path"`;

exports[`check win32 join [ 'some/nodejs\\\\windows', '../path' ] 1`] = `"some/nodejs/path"`;

exports[`check win32 join [ 'some\\\\windows\\\\only', '..\\\\path' ] 1`] = `"some/windows/path"`;

exports[`check win32 normalize '//\\\\windows\\\\..\\\\unix/mixed/' 1`] = `"/unix/mixed/"`;

exports[`check win32 normalize '/root/of/unix/somefile.ext' 1`] = `"/root/of/unix/somefile.ext"`;

exports[`check win32 normalize 'c:/windows/../nodejs/path' 1`] = `"c:/nodejs/path"`;

exports[`check win32 normalize 'c:/windows/nodejs/path' 1`] = `"c:/windows/nodejs/path"`;

exports[`check win32 normalize 'c:\\\\Windows\\\\Directory\\\\somefile.ext' 1`] = `"c:/Windows/Directory/somefile.ext"`;

exports[`check win32 normalize 'c:\\\\windows\\\\..\\\\nodejs\\\\path' 1`] = `"c:/nodejs/path"`;

exports[`check win32 normalize 'c:\\\\windows\\\\nodejs\\\\path' 1`] = `"c:/windows/nodejs/path"`;

exports[`check win32 parse '//\\\\windows\\\\..\\\\unix/mixed/' 1`] = `
Object {
  "base": "mixed",
  "dir": "/unix",
  "ext": "",
  "name": "mixed",
  "root": "/",
}
`;

exports[`check win32 parse '//windows\\\\unix/mixed' 1`] = `
Object {
  "base": "mixed",
  "dir": "\\\\\\\\windows\\\\unix\\\\",
  "ext": "",
  "name": "mixed",
  "root": "\\\\\\\\windows\\\\unix\\\\",
}
`;

exports[`check win32 parse '/root/of/unix/somefile.ext' 1`] = `
Object {
  "base": "somefile.ext",
  "dir": "/root/of/unix",
  "ext": ".ext",
  "name": "somefile",
  "root": "/",
}
`;

exports[`check win32 parse '\\\\windows//unix/mixed' 1`] = `
Object {
  "base": "mixed",
  "dir": "/windows/unix",
  "ext": "",
  "name": "mixed",
  "root": "/",
}
`;

exports[`check win32 parse 'c:/windows/../nodejs/path' 1`] = `
Object {
  "base": "path",
  "dir": "c:/nodejs",
  "ext": "",
  "name": "path",
  "root": "c:/",
}
`;

exports[`check win32 parse 'c:/windows/nodejs/path' 1`] = `
Object {
  "base": "path",
  "dir": "c:/windows/nodejs",
  "ext": "",
  "name": "path",
  "root": "c:/",
}
`;

exports[`check win32 parse 'c:\\\\Windows\\\\Directory\\\\somefile.ext' 1`] = `
Object {
  "base": "somefile.ext",
  "dir": "c:/Windows/Directory",
  "ext": ".ext",
  "name": "somefile",
  "root": "c:/",
}
`;

exports[`check win32 parse 'c:\\\\windows\\\\..\\\\nodejs\\\\path' 1`] = `
Object {
  "base": "path",
  "dir": "c:/nodejs",
  "ext": "",
  "name": "path",
  "root": "c:/",
}
`;

exports[`check win32 parse 'c:\\\\windows\\\\nodejs\\\\path' 1`] = `
Object {
  "base": "path",
  "dir": "c:/windows/nodejs",
  "ext": "",
  "name": "path",
  "root": "c:/",
}
`;
